vulnerabilities/importers/istio.py

```diff
@@ -0,0 +1,173 @@
+        for version in all_version_list:
+            if any([version in v for v in version_ranges]):
+                vuln_pkg_versions.append(version)
+
+        safe_pkg_versions = set(all_version_list) - set(vuln_pkg_versions)
+        return safe_pkg_versions, vuln_pkg_versions
+
+    def get_data_from_md(self, file):
+        data = {}
+        for line in file:
+            line = line.strip()
+            line = line.split()
+            if len(line) > 0 and line is not None:
+
+                start = line[0]
+
+                if start == "title:":
+                    data["title"] = " ".join(line[1:])
+                elif start == "description:":
+                    data["description"] = " ".join(line[1:])
+                elif start == "cves:":
+                    data["cves"] = " ".join(line[1:])
+                    data["cves"] = data["cves"].replace("[", "")
+                    data["cves"] = data["cves"].replace("]", "")
+                    data["cves"] = data["cves"].split(",")
+
+                elif start == "releases:":
+                    data["releases"] = " ".join(line[1:])
+                    data["releases"] = data["releases"].replace("[", "")
+                    data["releases"] = data["releases"].replace("]", "")
+                    data["releases"] = data["releases"].replace('"', "")
+                    data["releases"] = data["releases"].split(",")
+        releases = []
+        if data.get("releases"):
+            for release in data["releases"]:
+                release = release.strip()
+                release = release.split(" ")
+                if len(release) > 2:
+                    lbound = ">=" + release[0]
+                    ubound = "<=" + release[2]
+                    releases.append(lbound + "," + ubound)
+        data["releases"] = releases
+
+        return data
+
+    def process_file(self, path):
+
+        advisories = []
+
+        with open(path) as f:
+            data = {}
+
+            data = self.get_data_from_md(f)
+
+            if not data.get("cves"):
+                data["cves"] = [""]
+
+            for cve_id in data["cves"]:
+
+                if not cve_id.startswith("CVE"):
+                    continue
+
+                safe_pkg_versions = []
+                vuln_pkg_versions = []
+
+                if not data.get("releases"):
+                    data["releases"] = []
+
+                (
+                    safe_pkg_versions,
+                    vuln_pkg_versions,
+                ) = self.get_versions_for_pkg_from_range_list(data["releases"])
+
+                safe_purls = []
+                vuln_purls = []
+
+                cve_id = cve_id
+
+                safe_purls = {
+                    PackageURL(name="istio", type="golang", version=version)
+                    for version in safe_pkg_versions
+                }
+
+                vuln_purls = {
+                    PackageURL(name="istio", type="golang", version=version)
```

Comment:
Sure

Reply:
@pombredanne I am still unable to understand how to create two package types for a single purl. Can you explain it a bit.

Reply:
@tushar912 sorry if I was not clear: I meant to have multiple purls: one `pkg:/golang` and one `pkg:github` since the project has both personalities

Reply:
Oh Ok